lib/core/models/repositories/roleRepository.js

```diff
@@ -181,25 +194,130 @@ class RoleRepository extends Repository {
+   *
+   * @param {Role} role
+   */
+  checkRoleNativeRights(role) {
+    const roleRightsError = errorsManager.wrap('security', 'role');
+    Object.keys(role.controllers).forEach(roleController => {
+      if (roleController !== '*' && !this.kuzzle.funnel.controllers.has(roleController)) {
+        if (roleController.split('/').length > 1) {
+          // Meaning this is a plugin controller. So we skip it.
+          return;
+        }
+        roleRightsError.throw('unknown_controller', role._id, roleController);
+      }
+      if (roleController === '*') {
+        this.kuzzle.funnel.controllers.forEach(controller => {
+          Object.keys(role.controllers['*'].actions).forEach(action => {
+            if (action !== '*' && !controller._isAction(action)) {
+              roleRightsError.throw('unknown_controller_action', role._id, action, controller);
+            }
+          });
+        });
+      }
+      else {
+        const controller = this.kuzzle.funnel.controllers.get(roleController);
+        const actions = Object.keys(role.controllers[roleController].actions); 
+        actions.forEach(action => {
+          if (action !== '*' && !controller._isAction(action)) {
+            roleRightsError.throw('unknown_controller_action', role._id, action, roleController);
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Given a Role object, checks if its controllers and actions exist in plugins.
+   *
+   * @param {Role} role
+   * @param {Force} force
+   */
+  checkRolePluginsRights(role, force = false) {
+    const roleRightsError = errorsManager.wrap('security', 'role');
+    Object.keys(role.controllers).forEach(roleController => {
+      if (roleController.split('/').length === 1) {
+        // Meaning this is not a plugin controller. So we skip it.
+        return;
+      }
+      const pluginName = roleController.split('/')[0];
+      const pluginController = roleController.split('/')[1];
+      const availablePlugins = Object.keys(this.kuzzle.pluginsManager.plugins);
+      if (!availablePlugins.includes(pluginName)) {
+        if (!force) {
+          roleRightsError.throw('unavailable_plugin', role._id, pluginName);
+        }
+        this.kuzzle.log.warn(`[!WARNING!] Trying to set role "${role._id}" with plugin "${pluginName}" which doesn't exist or is not enabled.`);
+        return;
+      }
+      const roleActions = Object.keys(role.controllers[`${pluginName}/${pluginController}`].actions);
+      const pluginControllers = Object.keys(this.kuzzle.pluginsManager.plugins[pluginName].object.controllers);
+      if (pluginController !== '*' && !pluginControllers.includes(pluginController)) {
+        if (!force) {
+          roleRightsError.throw('unavailable_controller_plugin', role._id, pluginController, pluginName);
+        }
+        this.kuzzle.log.warn(`[!WARNING!] Trying to set role "${role._id}" with a non-existing controller "${pluginController}" in the plugin "${pluginName}".`);
```

Comment:
Misleading: it's not "trying" since it allows that creation, hence the warning.

```suggestion
        this.kuzzle.log.warn(`[!WARNING!] Role "${role._id}" gives access to the non-existing controller "${pluginController}" in the plugin "${pluginName}".`);
```